.github/workflows/pr-preview.yml

```diff
@@ -75,17 +79,38 @@ jobs:
           if [ "${BRANCH}" = "release/demo" ]
           then
             aws s3 cp "${WHEELFILE}" s3://core-previews/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl --acl public-read
-            echo -e "Wheel file download link: https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl"
+            S3_URL="https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl"
           else
             aws s3 cp "${WHEELFILE}" s3://core-previews/${PREVIEW_BRANCH}/ --acl public-read
-            echo -e "Wheel file download link: https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/${WHEELFILE}"
+            S3_URL="https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/${WHEELFILE}"
           fi
+          echo -e "Wheel file download link: ${S3_URL}"
 
           cd ../..
           # env variables don't carry over between gh action jobs
           echo "enable-setup=${{ env.AWS_ACCESS_KEY_ID != '' }}" >> $GITHUB_OUTPUT
           echo "preview-branch=$PREVIEW_BRANCH" >> $GITHUB_OUTPUT
-          echo "s3-url=https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/${WHEELFILE}" >> $GITHUB_OUTPUT
+          echo "s3-url=${S3_URL}" >> $GITHUB_OUTPUT
+
+      - name: "Set S3 URL as Github status"
+        if: success() && github.repository == 'streamlit/streamlit' && github.event_name == 'pull_request' && github.event.sender.type == 'User'
+        uses: actions/github-script@v6
+        env:
+          S3_URL: ${{ steps.exports.outputs.s3-url }}
+        with:
+          script: |
+            const { sha } = context.payload.pull_request.head
+            const { owner, repo } = context.repo
```

Comment:
super nit: think we can skip deconstruction of owner & repo since this only runs if we are in `streamlit/streamlit`? 
Then the POST request would look like `POST /repos/streamlit/streamlit/statuses/{sha}`

Reply:
Done.